src/check_cache.h

```diff
@@ -65,19 +62,15 @@ class CheckCache {
  private:
   class CacheElem {
    public:
-    CacheElem(const ::istio::mixer::v1::CheckResponse& response,
-              const int64_t time)
-        : check_response_(response), last_check_time_(time) {}
-
-    // Setter for check response.
-    inline void set_check_response(
-        const ::istio::mixer::v1::CheckResponse& check_response) {
-      check_response_ = check_response;
+    CacheElem(::google::protobuf::util::Status status, const int64_t time)
```

Comment:
The cache needs to capture the expiration count in addition to the expiration time. A given cache entry should only be used for the specified number of times.